distributed/scheduler.py

```diff
@@ -5144,13 +5164,11 @@ async def restart(self, client=None, timeout=30):
             )
         )
 
-        self.clear_task_state()
-
-        for plugin in list(self.plugins.values()):
-            try:
-                plugin.restart(self)
-            except Exception as e:
-                logger.exception(e)
+        if time() - start > timeout:
```

Comment:
How about gathering the shutdown of non-nanny workers and the restart of nannies in a single `wait_for`? This way, the timeout parameter would more closely reflect the actual timeout time of `restart()`.